src/amd/compiler/aco_live_var_analysis.cpp

```diff
@@ -70,6 +80,23 @@ void process_live_temps_per_block(Program *program, live& lives, Block* block, s
             sgpr_demand += it->size();
       }
    }
+=======
+   std::partition_copy(live_temps[block->index].begin(),
+                       live_temps[block->index].end(),
+                       std::inserter(live_sgprs, live_sgprs.end()),
+                       std::inserter(live_vgprs, live_vgprs.end()),
+                       [](Temp temp) { return temp.is_linear(); });
+
```

Comment:
I that faster? I'm asking because it would be really cool to get this algorithm faster. It is used multiple times in the backend...
Please fix the rebase artifacts (I know they are removed in a later commit, but they shouldn't appear in the first place).

Reply:
Yeah that one I am not sure about. I think that it is in the realm of "not measurably slower"

What it brings to the table is a clear separation of concerns and a more declarative style. I dont think that it matters with small blocks though I would have to measure it.